sklearn/ensemble/tests/test_iforest.py

```diff
@@ -211,3 +213,11 @@ def test_iforest_subsampled_features():
     clf = IsolationForest(max_features=0.8)
     clf.fit(X_train, y_train)
     clf.predict(X_test)
+
+def test_iforest_average_path_length():
+    # It tests non-regression for #8549 which used the wrong formula
+    # for average path length
+
+    assert_almost_equal(_average_path_length(1), 1., decimal=10)
+    assert_almost_equal(_average_path_length(5), 2.327020052, decimal=10)
```

Comment:
I think I'd rather this written out as:
`2*np.log(4) + 2 * np.euler_gamma − (2 * 4/5)`
unless you've got `2.327020052` straight from some reference table.